app/lib/analyzer/backend.dart

```diff
@@ -88,12 +90,15 @@ class AnalysisBackend {
         inserts.add(package);
       }
 
+      final DateTime prevTimestamp = version?.analysisTimestamp;
       if (version == null) {
         version = new PackageVersionAnalysis.fromAnalysis(analysis);
         inserts.add(version);
       } else if (version.updateWithLatest(analysis)) {
         inserts.add(version);
       }
+      final bool wasRace = prevTimestamp != null &&
+          version.analysisTimestamp.difference(prevTimestamp) < freshThreshold;
```

Comment:
If it was a race, why would we insert this one? We already know there is an up-to-date one stored, right?